checker/src/visitors.rs

```diff
@@ -1596,6 +1596,10 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
                     break;
                 }
             }
+            let elapsed_time_in_seconds = self.start_instant.elapsed().as_secs();
+            if elapsed_time_in_seconds >= k_limits::MAX_ANALYSIS_TIME_FOR_BODY {
+                return;
```

Comment:
Do you keep stats on any of these types of timeouts? Isn't it hard to track down problems when anything can just silently give up?

Reply:
Time-outs surface as WARN level entries in log output. I am painfully aware of any time-out.